### ISSUE TITLE
Bracer/Gloves/Weapon deflection recipes not working

### ISSUE THREAD

**Issue by vandervecken** [kevL/TccScripts#2]
The following recipes:

CATEGORY | Reagents | TAGS | Effects | CasterLevel
-- | -- | -- | -- | --
450 | “NW_IT_GEM005,cft_ess_earth2” | 2 | “1,1,0,0,0” | 8
450 | “NW_IT_GEM012,cft_ess_earth3” | 2 | “1,2,0,0,0” | 10
450 | “cft_ess_earth4,cft_gem_12” | 2 | “1,3,0,0,0” | 12

are set to use TAGs 2 (armor) when they should be using TAGs 11,1 - these are deflection AC bonus recipes designed for weapons and gloves. The Mage Armor versions of these recipes work fine, but these are the Shield of Faith recipes.

**Comment by vandervecken** [comment]
Aqvilinus argues that because we have Protection recipes +1-+5, that Deflection recipes intended for GLoves & Weapons (11,1) should instead be made to be just weapons (1).

**Comment by kevL** [comment]
don't have an opinion atm. Just want to point out that the type (deflection, natural, etc) of AC protection is dependent on the type of item that recieves the IP.

**Comment by Aqvilinus** [comment]
Tag (-2) is for amulets, rings, boots, belts, cloaks, bracers and helms.
As KevL said, the type of AC bonus depends on the type of an item (http://nwn.wikia.com/wiki/Armor_class#Armor_bonus - for NWN1, but the same is true for NWN2).
Amulets - natural armor bonus
Rings, belts, cloak, gloves, helms - deflection bonus
Boots - dodge bonus
Bracers - armor bonus

There's no way to make gloves granting other bonus than deflection. So Protection recipes offer the same benefits and are much better. That's why I made this proposion.

**Comment by kevL** [comment]
question, how does the engine distinguish between "gloves" and "bracers" during gameplay? Does it differentiate them, or are the two terms simply for human understanding?

**Comment by Aqvilinus** [comment]
I'm not sure I understand the question quite right. Bracers and gloves are different base types of items. Some of item properties (ex: damage and attack bonuses) don't work on bracers when unarmed.

**Comment by Aqvilinus** [comment]
Some of item properties (ex: damage and attack bonuses) don't work on bracers when fighting unarmed, but work on gloves. Gloves are treated by the engine more like weapon while bracers more like armor.

**Comment by kevL** [comment]
hmm, methinks this needs tweaking then,

'ginc_crafting'
int GetTccType(object oItem);

```c++
    case BASE_ITEM_GLOVES:
    case BASE_ITEM_BRACER: return TCC_TYPE_GLOVES;
```

ironically there doesn't seem to be any return for TCC_TYPE_WRISTS.

The funct is ofc used to match Tags with the baseitem-type that's in the crafting container (ie, when enchanting an item). A common Tag combination is "11,1" -- ie, TCC_TYPE_WRISTS or TCC_TYPE_MELEE. But if TCC_TYPE_WRISTS is never returned from GetTccType(oItem) then it strikes me that "11" will never get a match!

That Tag combination "11,1" can/could only match true melee weapons at present ...

If I do something, after closer examination, I'll try to keep that tag combination intact for the sake of Crafting.2da; i'd specify that enchanting an item of BASE_ITEM_GLOVES matches up with TCC_TYPE_WRISTS (aka "11"). Because gloves (not bracers) are the items used for attack, just like melee weapons.

That'd mean .. ah sh8t. That'd mean that TCC_TYPE_GLOVES (stress "GLOVES") are 36 (but should be 11, by the terminology). Perhaps I'll just refactor TCC_TYPE_GLOVES -> TCC_TYPE_BRACERS and TCC_TYPE_WRISTS -> TCC_TYPE_GLOVES ......

/needs coffee anyway

**Comment by vandervecken** [comment]
Perhaps we should make it so that Gloves are weapon-types, not wondrous item types. They have their own set of recipes related to weapons, and can't be given recipes related to generic Wondrous Items?

**Comment by kevL** [comment]
that makes sense,

also, until i/we figure out what to do with the TCC constant(s) gloves/wrists, focus on something else perhaps ... (up to you, it's not major it's just a thing)

**Comment by vandervecken** [comment]
I agree that they should be created using wondrous items, but the category of enchantments associated with each type are clearly different. Gloves should have a special subset of weapon enchantments that don't work on bracers, and vice versa. If there's an issue with creating them we should make sure we have blueprints available.

**Comment by kevL** [comment]
> Gloves should have a special subset of weapon enchantments that don't work on bracers, and vice versa.

i agree on that, and will be looking into separating TCC gloves from TCC bracers as time permits. I'll try to do it with as little potential damage to Crafting.2da as possible (but no guarantees atm)

**Comment by kevL** [comment]
in the toolset (item blueprints/templates) baseitem gauntlets (aka gloves) all have "Gloves" in their names and baseitem bracers all have "Bracers" in theirs.

that's good.

gloves have all sorts of IPs assigned, many of them combat-oriented but some having nothing to do with combat (think, Gloves of Appraisal, etc).

bracers, on the other hand, are mostly about bonus AC and bonus Dexterity, although there are other ips like bonus feats, immunities, cast spell, etc.

so far so good.

here's the rub:
rows 1159..1162 and 1257..1259 are MotB recipes for DexterityBonus +3..+9

but they're TAGGED "B36" baseitem type gloves -- that's not a big deal, it allows a crafter to enchant gloves instead of bracers w/ +Dex (note that there are other recipes that enchant TAGS=Any with +Dex, using different reagents and required Feat)

NOTE: I'm just blabbing trying to get this in my head, it's not big concerns.

but rows like these look a bit odd

![yata_crafting_dex4](https://user-images.githubusercontent.com/5395276/46900407-d3c5bf80-ce5e-11e8-8aa3-b1fac7241a85.png)

note for example the level difference

**Comment by vandervecken** [comment]
Yes they're odd. They're here:
https://gamefaqs.gamespot.com/pc/922154-neverwinter-nights-2/faqs/50274
The recipe on the top and bottom are explicitly based on Mask of the Betrayer.
The top one is for high level crafting with essences only. At that level you don't even need a gem.
The bottom one is for Spirit Essences with Mold Spirit - a recipe only the soul eater can perform.
The middle one is the standard recipe.

I don't think this is something worth worrying about.

I do think it's notable that Gloves have certain favoured non-combat recipes, like Dex and Appraisal. We shouldn't mess with that. I'm not sure if that means we should allow all wondrous item recipes on gloves though.

On another note, I'm away for much of this weekend, so will be in touch sporadically.

**Comment by kevL** [comment]
ok/ ok

am going to do some refactoring of the TCC_TYPE_* constants and see what jumps out as worthwhile,

**Comment by kevL** [comment]
Can someone rephrase what the issue is here? Seems to be something about

a) Shield of Faith recipes not working, and
b) redundant Deflection recipes

**Comment by kevL** [comment]
well, just to be clear ... the code should now handle TAGS
"36" - gloves exclusively
"78" - bracer exclusively
"11" - gloves or bracer (as before)

But as to recipes ... uh ... this is what i'd do if I had the time to invest

ensure there are recipes +1 to +8 (or just +5 /shrug) that can be applied to items of types that grant AC of each AC-type (dodge, natural, armor, shield, deflection) -- whether crafter is using MageArmor/ImprovedMageArmor, Shield (might want to limit Shield triggers to recipes that apply shield AC only), or Shield of Faith ... etc

But if things are reasonably okay as-is, leave things as-is (until/unless you run into a limitation that you don't like yourself, or someone issues a more explicit request like "hey, Ima cleric and can't use Shield of Faith to add a Dodge bonus to my boots")

because, well you know, there's gonna be a lot of permutations to balance out, restricting some recipes to certain items while adding recipes/triggers for other item-types ...

**Comment by vandervecken** [comment]
Here's what I will do:
Walk through every recipe that uses (or should use) 11, and split it to 36 or 78 if necessary, then call it a day. My goal was never really to expand the recipe list beyond filling in obvious holes.

**Comment by kevL** [comment]
sounds ok :)

**Comment by vandervecken** [comment]
Urgh, do we need new TCC molds for making bracers, as well as the UTIs for unenchanted bracers and the relevant recipes?
Otherwise, how is a user going to create a Bracer for the now bracer-only recipes?

**Comment by kevL** [comment]
idk but the code supports it ...

or did i break something ...

**Comment by vandervecken** [comment]
That's not quite what I mean. The point is that if we differentiate Bracers from Gloves in the actual recipes, then I will need to provide a way for the user to create Bracers. If I don't, how are they going to craft the Bracer set of their dreams?
So now I need:
* TCC Bracer mold
* UTIs for Bracers and Masterwork bracers
* Possibly UTIs for bracers made of Mithral and Adamantite, and any other materials, along with the masterwork versions
* Recipes for blacksmiths for creating these bracers

on top of everything.

I'm tempted to leave this for a hypothetical version 2.0.6. What do you think?

**Comment by kevL** [comment]
y, leave it for the next push ...

it seems like your current mindframe is on recipes, but this sounds like it needs a lot of focus on various .UTIs and their interaction/organization

**Comment by vandervecken** [comment]
I do think things will stand out for gloves vs Bracers. Armor bonuses on Bracers, attack bonuses on gloves, for example.

But the reason I want to hold off on this is because it's the only major restricting change we've explored. Things that used to be possible won't be possible any more (like putting Armor bonuses on Gloves of Dexterity or putting Dexterity bonuses on Bracers).
And I want to give players a clean major-bugfix version, 2.0.5, before I make major changes.
So that means the only thing left for now is the adamantine shields thing.

**Comment by kevL** [comment]
makes sense,

given the (future) restrictive change, i'd bump that (future) version to, say, 2.1
